test/api.jl

```diff
@@ -115,5 +116,29 @@ recommendation = provide_recommendation(trial_vector)
 recommendation_multi = provide_recommendation(trial_vector_multihyperparams)
 @test recommendation_multi == Dict(:x => 3, :y => 7)
 
+# Test case reproducing the issue in “obs_memo” function from the higher level
+# More details on this issue see https://github.com/IQVIA-ML/TreeParzen.jl/issues/86
+space = Dict(
+    :a => HP.Choice(:a, [8, 9, 10]),
+    :b => HP.Choice(:b, [1, 2, 3])
+    )
+
+Random.seed!(10)
+trials = TreeParzen.Trials.Trial[]
+
+for i in 1:30
+    trial = ask(space, trials, TreeParzen.Config())
+    tell!(trials, trial, trial.hyperparams[:b]/trial.hyperparams[:a])
+end
+# With this example space and loss function, expected vals for the best param should be different
+# but with the issue in “obs_memo” function, they are equal due to same nid
+@test_broken trials[30].vals[:a] != trials[30].vals[:b]
+
+# To have the smaller loss, the expected best hyperparams[:a] is 10 and hyperparams[:b] is 1
+expected_a = 10
+expected_b = 1
+
+@test_broken (trials[30].hyperparams[:a] == expected_a) && (trials[30].hyperparams[:b] == expected_b)
```

Comment:
To make this more robust, instead of putting expected a, you want this to be from the set of values for distributions A/B respectively.

Reply:
```julia
@test_broken trials[30].hyperparams[:a] in a_set
@test_broken trials[30].hyperparams[:b] in b_set
```

Reply:
re "you want this to be from the set of values for distributions A/B respectively" : No, is not a set of values we expected 10 for a and 1 for b. Please check this python code, it might help to clarify this issue 
```
from hyperopt import Trials, hp, tpe, fmin
from hyperopt.fmin import generate_trials_to_calculate
from hyperopt import space_eval

spaceVar = {'a' : hp.choice('a', [8, 9, 10]),
            'b' : hp.choice('b', [1, 2, 3])}
def objective(space):
    print (space)
    return float(space['b'])/float(space['a'])

best = fmin(fn=objective, space=spaceVar, algo=tpe.suggest, max_evals=30)
space_eval(spaceVar, best)
```

Reply:
There's no specific reason to expect **exactly** 10 and **exactly** 1, however, there is a reason to expect that all of the sampled values for A is from [8, 9, 10] and all of the sampled values for B is from [1, 2, 3].